internal/ratelimit/ratelimit_test.go

```diff
@@ -0,0 +1,664 @@
+			Reset:     10,
+		}
+		manager.activeRequestsMu.Lock()
+		manager.activeRequests.Store(0)
+		manager.activeRequestsMu.Unlock()
+		delay := manager.calculateDelay(rateLimitInfo)
+
+		// Base delay: (10*0.8)/5 = 1.6s, with jitter: 1.76-1.92s
+		assert.Greater(t, delay, 1*time.Second)
+		assert.Less(t, delay, time.Duration(2.5*float64(time.Second)))
+	})
+
+	t.Run("long reset time should be capped", func(t *testing.T) {
+		// One remaining request, long reset time (should be capped at 30s)
+		rateLimitInfo := &RateLimitInfo{
+			Limit:     100,
+			Remaining: 1,
+			Reset:     60,
+		}
+		manager.activeRequestsMu.Lock()
+		manager.activeRequests.Store(0)
+		manager.activeRequestsMu.Unlock()
+		delay := manager.calculateDelay(rateLimitInfo)
+
+		// Base calculation: (60*0.8)/1 = 48s, but capped at 30s, with jitter: 33-36s
+		assert.Greater(t, delay, 30*time.Second)
+		assert.Less(t, delay, 37*time.Second)
+	})
+}
+
+func TestCalculateDelayWithActiveRequestScaling(t *testing.T) {
+	manager := NewManager(nil)
+
+	// Some remaining requests, with active requests
+	rateLimitInfo := &RateLimitInfo{
+		Limit:     100,
+		Remaining: 5,
+		Reset:     10,
+	}
+	manager.activeRequestsMu.Lock()
+	manager.activeRequests.Store(2)
+	manager.activeRequestsMu.Unlock()
+	delay := manager.calculateDelay(rateLimitInfo)
+
+	// Base delay: 1.6s, scaling: 1.6*(1+2*0.2) = 2.24s, with jitter: 2.46-2.69s
+	assert.Greater(t, delay, 2*time.Second)
+	assert.Less(t, delay, time.Duration(3.5*float64(time.Second)))
+}
+
+func TestActiveRequestsCleanup(t *testing.T) {
+	// Create manager with short cleanup interval for testing
+	cleanupInterval := 100 * time.Millisecond
+	manager := NewManager(&ManagerOptions{
+		CleanupInterval: &cleanupInterval,
+	})
+	defer manager.Stop()
+
+	// Artificially set activeRequests to simulate stuck state
+	// Use write lock to ensure exclusive access during setup
+	manager.activeRequestsMu.Lock()
+	manager.activeRequests.Store(5)
+	stuckCount := manager.activeRequests.Load()
+	manager.activeRequestsMu.Unlock()
+
+	assert.Equal(t, int64(5), stuckCount, "activeRequests should be set to 5")
+
+	// Wait for cleanup to run (should happen within 100ms + some buffer)
+	// Use a longer wait to ensure cleanup has definitely run
+	time.Sleep(300 * time.Millisecond)
+
+	// Check that activeRequests has been reset
+	// Use read lock to safely check the value
+	manager.activeRequestsMu.RLock()
+	cleanedCount := manager.activeRequests.Load()
+	manager.activeRequestsMu.RUnlock()
+
+	assert.Equal(t, int64(0), cleanedCount, "activeRequests should be cleaned up to 0")
+}
+
+func TestNewManagerWithOptions(t *testing.T) {
+	t.Run("with nil options uses defaults", func(t *testing.T) {
+		manager := NewManager(nil)
+		assert.NotNil(t, manager)
+		assert.NotNil(t, manager.logger)
+		assert.Equal(t, 0.02, manager.threshold)
+		assert.Equal(t, 50*time.Millisecond, manager.minRequestInterval)
+		assert.Equal(t, 30*time.Second, manager.cleanupInterval)
+	})
```

Comment:
I don't think this is needed. But if you do test the defaults, please use `const`-s for the hard coded values. Or at least call the function that creates a default options struct.